great_expectations/rule_based_profiler/data_assistant/data_assistant.py

```diff
@@ -105,6 +109,13 @@ def __init__(
         )
         self._build_profiler()
 
+    @classmethod
+    def __init_subclass__(cls) -> None:
+        if not hasattr(cls, "__alias__"):
+            raise NotImplementedError(
+                f"Class {cls} lacks required `__alias__` class attribute of type Optional[str]"
+            )
```

Comment:
In order for users to be able to define aliases, they need to be able to configure the value through their own custom implementations of the assistant. As such, we should place the burden of this responsibility within the assistant (as opposed to the dispatcher)

Reply:
I went with a dunder attribute here instead of a property because at the time of usage, we don't have class instance (just the types). I could use an abstract class or static method but opted for a slightly different approach.

Reply:
As it stands, you NEED to always define an `__alias__` attr (even if you're just setting to `None`). This may not be the cleanest way to do it but I prefer explicitness to convey intent.

Reply:
I'm not clear on the semantics of a dunder method and if it applies here. Can you explain your thinking here?

Aliases are not required. The way to think of it is as a second name under which the assistant is registered. If there is no alias, then the assistant is registered under just one name. The program should not explode if Alias is missing

Reply:
Hmm so dunder ("double underscore") attributes are just convention when defining something that has a special purpose. Python's stdlib often uses them to define core language features (`__init__`, `__new__`, etc).

Some libraries outside of the stdlib, like SQLAlchemy, will make use of them:
```python
base = declarative_base()

class MyModel(Base):
    __table__ = "my_model" # Let's our ORM know what to access within the DB
   ...
```

If there's any confusion, I think it's safest to go with a private attribute (`_alias`).

Reply:
In regards to your point about alias exploding, I definitely agree. I can circumvent this by adding a `_alias` attribute to the base class and setting it to `None` so that child classes will pass the check.

We can then get rid of this `__init_subclass__` check and always assume we have the attribute available when iterating through assistant types.